modules/caddyhttp/server.go

```diff
@@ -82,6 +82,12 @@ type Server struct {
 	// HTTP request headers.
 	MaxHeaderBytes int `json:"max_header_bytes,omitempty"`
 
+	// Enable full-duplex communication for HTTP/1.1 requests.
+	// Only has an effect if Caddy was built with Go 1.21 or later.
```

Comment:
Should we mention that this can brick/hang clients if they don't support this?

Reply:
Probably. But I'm not sure I understand well the circumstances where it can happen.

Reply:
What if we say something like: "Test thoroughly with your client, as some clients may not support full-duplex HTTP/1.1 which can cause them to deadlock."

Reply:
Done, also pulled in the text from Go stdlib which clarifies what it does.